tests/forms_tests/field_tests/test_urlfield.py

```diff
@@ -151,3 +151,13 @@ def test_urlfield_not_string(self):
     def test_urlfield_normalization(self):
         f = URLField()
         self.assertEqual(f.clean('http://example.com/     '), 'http://example.com/')
+
+    def test_urlfield_strip_false(self):
+        f = URLField(strip=False)
+        with self.assertRaisesMessage(ValidationError, "'Enter a valid URL.'"):
+            f.clean('http://example.com ')
```

Comment:
@MarkusH Before the changes this test would fail. Instead of raising a ValdiationError, an AttributeError ('NoneType' object has no attribute 'strip') would be thrown, as a strip() call was still made (on the None value) even though strip=False was set.

In other words -> this is the regression test.